### ISSUE TITLE
Editing properties

### ISSUE THREAD

**Issue by dieguico** [karimbahgat/PyGeoj#1]
Hi Karim!

First of all. Thank you for your work. :)

I can't edit the properties of the features:

>print testfile[0].properties
{u'ALTA_DB': u'2014/10/15', u'ID_CP': 81490000001.0, u'COD_POSTAL': u'08516'}
>testfile[0].properties={u'zipcode':u'08021'}
>print testfile[0].properties
{u'ALTA_DB': u'2014/10/15', u'ID_CP': 81490000001.0, u'COD_POSTAL': u'08516'}

*while I can do other edits*

>print testfile.bbox
[0.159181151588882, 0.310456711008385, 42.82497667423411, 42.861370527827546]
>testfile.bbox=[0,0,0,0]
>print testfile.bbox
[0,0,0,0]

Any idea?

Thanks!

Diego

**Comment by karimbahgat** [comment]
Hi Diego,

Thanks for pointing out this problem. 

It appears that editing features in place doesnt work in the current version. This is because all features are actually stored as dicts, and are only copied into a feature instance upon indexing or looping. Changing the feature only changes the copy, not the original. You will have to replace the orig feat with the changed one to make it work. 

This is not ideal of course, so i will try making a new version where features are stored as instances so they can be edited. 

If you use pygeoj alot, feel free to tell me of any other changes or suggestions to make it easier to use, eg how to delete features etc. 

Karim

**Comment by dieguico** [comment]
Hi Karim,

I can add items to the properties dict:
>for feature in testfile:
>    feature.properties[u'zipcode']=feature.properties['COD_POSTAL']

I can delete items from the properties dict:
>for feature in testfile:
>    del feature.properties['COD_POSTAL']

I.e., in my case it works doing this:
>for feature in testfile:
>    del feature.properties['ID_CP']
>    del feature.properties['ALTA_DB']
>    feature.properties['zipcode']=feature.properties['COD_POSTAL']
>    del feature.properties['COD_POSTAL']

But I can't assign a new dictionary directly 
>for feature in testfile:
>    feature.properties={u'zipcode': feature.properties['COD_POSTAL']}

Best,

Diego

**Comment by karimbahgat** [comment]
I see. I will have to clean up the internals and switch to storing each feature as a feature instance. With a better overview and cleaner approach it should be straightforward to assign a new dictionary and do any other changes. Will let you know.

**Comment by Tcboo** [comment]
Hi Karim,
I was using the old version of the module (awesome stuff!) and just updated to the new release.  I noticed that the .getfeature method is not working.
Thanks,
Sarah

**Comment by karimbahgat** [comment]
Hi @Tcboo , sorry for the delayed response. 

I just released a new bugfix version 0.22 on pypi that clarifies in the readme which methods to use. Specifically I have changed the getfeature, addfeature, replacefeature, and removefeature to have underscore between the words, such as get_feature, replace_feature, etc. I don't want to break yours or others old scripts, so I have also brought back the original syntax purely for legacy use, and added a "will be depreceated" warning. 

@dieguico I also fixed the internal bug in the new version, so that setting with feat.geometry = ... or feat.property = ... should work. Though I kept the dictionary data structure, meaning that Feature and Geometry instances are simply wrappers around the underlying data. 

Karim
